### PR TITLE
issue/183 concat_ws 

### PR DESCRIPTION
**Pull request by pjazdzewski1990**
For issue https://github.com/zio/zio-sql/issues/183

### CODE REVIEW

**Comment by jczuchnowski** [comment]
I think this is good enough. We need to figure out how to handle vararg functions but it can be done some other time.

File: core/jvm/src/main/scala/zio/sql/expr.scala [hunk 1]
```diff
@@ -108,6 +108,7 @@ trait ExprModule extends NewtypesModule with FeaturesModule with OpsModule {
 
     def typeTagOf[A](expr: Expr[_, _, A]): TypeTag[A] = expr.asInstanceOf[InvariantExpr[_, _, A]].typeTag
 
+    implicit def literal(str: String): Expr[Features.Literal, Any, String] = Expr.Literal(s"'${str}'")
```
Comment (jczuchnowski) [inline core/jvm/src/main/scala/zio/sql/expr.scala:111]: Please remove this. We need to deal with at the `renderRead` level.

File: core/jvm/src/main/scala/zio/sql/expr.scala [hunk 2]
```diff
@@ -279,6 +280,9 @@ trait ExprModule extends NewtypesModule with FeaturesModule with OpsModule {
     val Ascii       = FunctionDef[String, Int](FunctionName("ascii"))
     val CharLength  = FunctionDef[String, Int](FunctionName("character_length"))
     val Concat      = FunctionDef[(String, String), String](FunctionName("concat"))
+    val ConcatWs2   = FunctionDef[(String, String), String](FunctionName("concat_ws"))
```
Comment (jczuchnowski) [inline core/jvm/src/main/scala/zio/sql/expr.scala:283]: So, `concat_ws` is not available at least for Oracle, so we need to do it in the db-specific modules. For now it's fine to do it in PostgresModule, although SQL Server also supports this function.

**Comment by jczuchnowski** [comment]
@pjazdzewski1990 couple of problems after all

File: core/jvm/src/main/scala/zio/sql/expr.scala (hunk 1 above)
Comment (jczuchnowski) [reply core/jvm/src/main/scala/zio/sql/expr.scala:111]: You can add single quotes to the parameters in tests. We'll fix it once we fix `renderRead`.

Comment (pjazdzewski1990) [reply core/jvm/src/main/scala/zio/sql/expr.scala]: Done

File: core/jvm/src/main/scala/zio/sql/expr.scala (hunk 2 above)
Comment (pjazdzewski1990) [reply core/jvm/src/main/scala/zio/sql/expr.scala:292]: Ok, in another PR

**Review by jczuchnowski — APPROVED** [review]

**Pull request merged**